onnx/defs/math/defs.cc

```diff
@@ -416,6 +471,117 @@ will throw errors.
   .TypeConstraint("T", { "tensor(float16)", "tensor(float)", "tensor(double)" },
         "Constrain input and output types to float tensors.");
 
+OPERATOR_SCHEMA(LogSoftmax)
+  .NumInputs(1)
+  .NumOutputs(1)
+  .SetDoc(R"DOC(
+The operator computes the log softmax normalized values for each layer in the batch
+ of the given input. The input is a 2-D tensor (Tensor<float>) of size
+(batch_size x input_feature_dimensions). The output tensor has the same shape
+and contains the softmax normalized values of the corresponding input.
+
+X does not need to explicitly be a 2D vector; rather, it will be
```

Comment:
how about to avoid copy-pasting this chunk we concat strings? There's FillUsing trick (search repo for it) or simple concat might work too.